File: tests/fixtures/cpp/user_service.cpp
#include "user_service.hpp"

namespace app {

UserService::UserService(UserRepository repo, std::string name)
    : repo_(repo), name_(name) {}

std::string UserService::get_name() const {
    return name_;
}

std::string UserService::lookup(int id) const {
    return repo_.find(id);
}

size_t UserService::total_users() const {
    return repo_.count();
}

std::string UserService::greet() const {
    return "Hello, " + get_name();
}

const char* UserService::classify(int count) const {
    if (count > 10) {
        return "many";
    } else if (count > 0) {
        return "few";
    }
    return "none";
}

UserService make_service(std::string name) {
    UserRepository repo;
    return UserService(repo, name);
}

} // namespace app

File: tests/fixtures/rust/user_service.rs
use crate::user_repository::UserRepository;

pub struct UserService {
    repo: UserRepository,
    name: String,
}

impl UserService {
    pub fn new(repo: UserRepository, name: String) -> Self {
        UserService { repo, name }
    }

    pub fn get_name(&self) -> &str {
        &self.name
    }

    pub fn lookup(&self, id: u32) -> Option<String> {
        self.repo.find(id).cloned()
    }

    pub fn total_users(&self) -> usize {
        self.repo.count()
    }

    pub fn greet(&self) -> String {
        format!("Hello, {}", self.get_name())
    }

    pub fn classify(&self, count: i32) -> &'static str {
        if count > 10 {
            "many"
        } else if count > 0 {
            "few"
        } else {
            "none"
        }
    }
}

pub fn make_service(name: String) -> UserService {
    let repo = UserRepository::new();
    UserService::new(repo, name)
}
